http.go

```diff
@@ -43,9 +43,14 @@ type ServerHandler struct {
 }
 
 func (sh *ServerHandler) ServeHTTP(response http.ResponseWriter, request *http.Request) {
-	debugLog := log.WithPrefix(sh.Logger, level.Key(), level.DebugValue())
-	infoLog := log.WithPrefix(sh.Logger, level.Key(), level.InfoValue())
-	errorLog := log.WithPrefix(sh.Logger, level.Key(), level.ErrorValue())
+	logger := logging.GetLogger(request.Context())
+	if logger == nil {
+	    logger = sh.Logger
+	}
```

Comment:
```suggestion
```

`logging.GetLogger()` can never return `nil`. Having this check isn't a problem, but it leaves an untestable path in terms of unit test coverage. 😅

Reply:
I have changed it to logging.DefaultLogger() instead of nil in the latest commit.

Reply:
@kristinapathak  now we are comparing it is the default logger instead of nil, we haven't updated  if else block because we don't want to use the default logger which is provided by logging package.

Here also we did the same thing https://github.com/xmidt-org/argus/pull/145
Please refer to last conversation